src/graph.rs

```diff
@@ -86,6 +92,131 @@ impl<'tree> Graph<'tree> {
 
         DisplayGraph(self, ctx)
     }
+
+    pub fn display_json<'a>(&'a self, ctx: &'a Context) -> () {
+        // TODO: move this all into a new module
```

Comment:
This is really complicating (not to mention bloating) this file, so I'm likely to move it elsewhere.

Reply:
@robrix I think I left that thumbs down accidentally, I have no strong feelings